src/fastga_he/gui/power_train_network_viewer.py

```diff
@@ -57,65 +207,449 @@ def power_train_network_viewer(
+        +-----+-----+-----+-----+-----+
+        |     |     |     |     |     |
+        +-----+-----+-----+-----+-----+
+        |     |     |  M  |     |     |
+        +-----+-----+-----+-----+-----+
+        |  L  |     |  C  |     |  R  |
+        +-----+-----+-----+-----+-----+
+        |     |     |  B  |     |     |
+        +-----+-----+-----+-----+-----+
+
+        color_icon_urls: List of URLs for color icons
+        legend_scaling: Scaling factor for the legend size
+    """
+
+    if len(legend_position) != 2:
+        legend_scaling = "TR"
 
-    os.chdir(directory_to_save_graph)
+    if "T" in legend_position:
+        legend_y_start = 600
+    elif "B" in legend_position:
+        legend_y_start = 50
+    elif "M" in legend_position:
+        legend_y_start = 325
+    else:
+        legend_y_start = 600
 
-    net.from_nx(graph)
-    net.show(graph_name)
+    if "R" in legend_position:
+        legend_x_start = 500
+    elif "L" in legend_position:
+        legend_x_start = -50
+    elif "C" in legend_position:
+        legend_x_start = 225
+    else:
+        legend_x_start = 500
+
+    # Legend items
+    legend_items = [
+        (0, "Fuel Flow"),
+        (1, "Mechanical Power"),
+        (2, "Electrical Current"),
+    ]
+
+    legend_item_height = int(22 * legend_scaling)
+    legend_item_start_y = legend_y_start - int(25 * legend_scaling)
+
+    for i, (icon_idx, description) in enumerate(legend_items):
+        y_position = legend_item_start_y - (i * legend_item_height)
+
+        # Create data source for icon
+        icon_source = bkmodel.ColumnDataSource(
+            data=dict(
+                x=[legend_x_start + int(10 * legend_scaling)],
+                y=[y_position],
+                url=[color_icon_urls[icon_idx]],
+            )
+        )
+
+        # Draw color icon image
+        plot.image_url(
+            url="url",
+            x="x",
+            y="y",
+            w=9 * legend_scaling,
+            h=12 * legend_scaling,
+            anchor="center",
+            source=icon_source,
+        )
+
+        # Add text label next to the color icon image
+        label_source = bkmodel.ColumnDataSource(
+            data=dict(
+                x=[legend_x_start + 25],
+                y=[y_position],
+                text=[description],
+            )
+        )
+
+        labels = bkmodel.LabelSet(
+            x="x",
+            y="y",
+            text="text",
+            source=label_source,
+            text_align="left",
+            text_baseline="middle",
+            text_color="white",
+            text_font_size=str(10 * legend_scaling) + "pt",
+        )
+        plot.add_layout(labels)
+
+
+def _string_clean_up(old_string):
+    """
+    Clean up list content for better readability.
+    """
+    # In case for list type definition
+    if isinstance(old_string, list):
+        old_string = old_string[0].capitalize() + ", " + old_string[1].capitalize()
```

Comment:
So this works only for list with two elements ? There is a wa to make it more general, like using `.join()`